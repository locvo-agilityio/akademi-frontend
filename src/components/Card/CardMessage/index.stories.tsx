import { StoryObj, Meta } from '@storybook/react';

import CardMessage from '.';
import { Box } from '@chakra-ui/react';

const meta = {
  title: 'Components/Cards/CardMessage',
  component: CardMessage,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
  render: (args) => (
    <Box w="400px">
      <CardMessage {...args} />
    </Box>
  ),
} satisfies Meta<typeof CardMessage>;

export default meta;

type Story = StoryObj<typeof CardMessage>;

export const Default: Story = {
  args: {
    name: 'John Doe',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '10:00',
  },
};
