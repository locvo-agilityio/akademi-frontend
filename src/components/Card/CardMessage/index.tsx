import { memo } from 'react';
import {
  Avatar,
  Card,
  CardBody,
  CardHeader,
  Flex,
  Text,
  VStack,
} from '@chakra-ui/react';

interface CardMessageProps {
  name: string;
  content: string;
  avatar?: string;
  time: string;
}

const CardMessage = ({
  name,
  content,
  avatar = '',
  time,
}: CardMessageProps) => (
  <Card
    p={5}
    w="full"
    display="flex"
    flexDirection="row"
    gap={4}
    boxShadow="none"
    borderBottom="1px solid"
    borderColor="lightGray"
    borderRadius="none"
  >
    <CardHeader p={0} display="flex" gap={6}>
      <VStack>
        <Avatar size="md" name={name} src={avatar} />
      </VStack>
    </CardHeader>

    <CardBody
      p={0}
      display="flex"
      flexDirection="column"
      alignItems="left"
      gap={1}
    >
      <Flex justifyContent="space-between">
        <Text as="p" fontWeight="bold" color="darkblue">
          {name}
        </Text>
        <Text as="span" color="gray.600">
          {time}
        </Text>
      </Flex>
      <Text as="span" color="gray.600" whiteSpace="break-spaces" noOfLines={1}>
        {content}
      </Text>
    </CardBody>
  </Card>
);

export default memo(CardMessage);
