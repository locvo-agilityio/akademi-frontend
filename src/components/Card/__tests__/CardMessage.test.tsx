import { render } from '@testing-library/react';

// Components
import CardMessage from '../CardMessage';

describe('CardMessage component', () => {
  const mockProps = {
    name: 'John Doe',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    time: '10:00',
  };

  it('should render correctly with default props', () => {
    const { container } = render(<CardMessage {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
