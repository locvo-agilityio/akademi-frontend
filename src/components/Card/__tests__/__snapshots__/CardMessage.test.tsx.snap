// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMessage component should render correctly with default props 1`] = `
<div>
  <div
    class="chakra-card css-13wq1zv"
  >
    <div
      class="chakra-card__header css-17udfxa"
    >
      <div
        class="chakra-stack css-tl3ftk"
      >
        <span
          class="chakra-avatar css-1ruhx4w"
        >
          <div
            aria-label="John Doe"
            class="chakra-avatar__initials css-0"
            role="img"
          >
            JD
          </div>
        </span>
      </div>
    </div>
    <div
      class="chakra-card__body css-4xanpx"
    >
      <div
        class="css-gg4vpm"
      >
        <p
          class="chakra-text css-djt7re"
        >
          John Doe
        </p>
        <span
          class="chakra-text css-djevj3"
        >
          10:00
        </span>
      </div>
      <span
        class="chakra-text css-1hmu7y7"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </span>
    </div>
  </div>
</div>
`;
